nucleus/payara-modules/payara-executor-service/src/main/java/fish/payara/nucleus/executorservice/PayaraExecutorService.java

```diff
@@ -117,93 +117,94 @@ public void event(Event event) {
             scheduledThreadPoolExecutor.shutdown();
         }
     }
-    
+
     private void initialiseThreadPools() {
         int threadPoolExecutorQueueSize = Integer.valueOf(
                 payaraExecutorServiceConfiguration.getThreadPoolExecutorQueueSize());
         if (threadPoolExecutorQueueSize > 0) {
             threadPoolExecutor = new ThreadPoolExecutor(
-                    Integer.valueOf(payaraExecutorServiceConfiguration.getThreadPoolExecutorCorePoolSize()), 
-                    Integer.valueOf(payaraExecutorServiceConfiguration.getThreadPoolExecutorMaxPoolSize()), 
-                    Integer.valueOf(payaraExecutorServiceConfiguration.getThreadPoolExecutorKeepAliveTime()), 
+                    Integer.valueOf(payaraExecutorServiceConfiguration.getThreadPoolExecutorCorePoolSize()),
+                    Integer.valueOf(payaraExecutorServiceConfiguration.getThreadPoolExecutorMaxPoolSize()),
+                    Integer.valueOf(payaraExecutorServiceConfiguration.getThreadPoolExecutorKeepAliveTime()),
                     TimeUnit.valueOf(payaraExecutorServiceConfiguration.getThreadPoolExecutorKeepAliveTimeUnit()),
-                    new LinkedBlockingQueue<>(threadPoolExecutorQueueSize), 
+                    new LinkedBlockingQueue<>(threadPoolExecutorQueueSize),
                     (Runnable r) -> new Thread(r, "payara-executor-service-task"));
         } else {
             threadPoolExecutor = new ThreadPoolExecutor(
-                    Integer.valueOf(payaraExecutorServiceConfiguration.getThreadPoolExecutorCorePoolSize()), 
-                    Integer.valueOf(payaraExecutorServiceConfiguration.getThreadPoolExecutorMaxPoolSize()), 
-                    Integer.valueOf(payaraExecutorServiceConfiguration.getThreadPoolExecutorKeepAliveTime()), 
+                    Integer.valueOf(payaraExecutorServiceConfiguration.getThreadPoolExecutorCorePoolSize()),
+                    Integer.valueOf(payaraExecutorServiceConfiguration.getThreadPoolExecutorMaxPoolSize()),
+                    Integer.valueOf(payaraExecutorServiceConfiguration.getThreadPoolExecutorKeepAliveTime()),
                     TimeUnit.valueOf(payaraExecutorServiceConfiguration.getThreadPoolExecutorKeepAliveTimeUnit()),
                     new SynchronousQueue<>(), (Runnable r) -> new Thread(r, "payara-executor-service-task"));
         }
 
         scheduledThreadPoolExecutor = new ScheduledThreadPoolExecutor(
                 Integer.valueOf(payaraExecutorServiceConfiguration.getScheduledThreadPoolExecutorCorePoolSize()),
                 (Runnable r) -> new Thread(r, "payara-executor-service-scheduled-task"));
+        scheduledThreadPoolExecutor.setRemoveOnCancelPolicy(true);
```

Comment:
I think this should be configurable on the service

Reply:
Also I think the old purge on x number of tasks should be also retained in case there is a performance hit on removing the tasks for many small and rapid transactions like in high volume web applications with timeouts enabled.

Reply:
Removed config from PayaraExecutorService